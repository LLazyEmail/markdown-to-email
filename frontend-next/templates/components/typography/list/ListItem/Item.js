import React from 'react';
import { styles } from './styles';

const ListItem = ({ children }) => (
  <li style={styles.li}>
    {/* was <p></p> */}
    <div dir="ltr" role="presentation" style={styles.presentation} data-testid="listItemTest">
      {children}
    </div>
  </li>
);

export default ListItem;
