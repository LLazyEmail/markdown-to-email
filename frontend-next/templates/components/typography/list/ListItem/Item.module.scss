li { 
    mso-line-height-rule: exactly;
    -ms-text-size-adjust: 100%;
    -webkit-text-size-adjust: 100%;
  }
.presentation {
  line-height: 125%;
  margin: 10px 0;
  padding: 0;
  mso-line-height-rule: exactly;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  color: #111111;
  font-family: "Source Sans Pro", "Helvetica Neue", Helvetica, Arial, sans-serif;
  font-size: 18px;
  text-align: left;
}