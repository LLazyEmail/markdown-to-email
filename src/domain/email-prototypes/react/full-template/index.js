import { generateReactFullTemplate } from '../../../../parserMDReact/methods';

// import {
//   generateReactContent,
// } from './parserMDReact';


export {
    generateReactFullTemplate
}