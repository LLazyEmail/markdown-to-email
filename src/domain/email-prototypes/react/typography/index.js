import headingLiteral from './literals/heading';
import imageLiteral from './literals/image';
import italicLiteral from './literals/italic';
import linkLiteral from './literals/link';
import listLiteral from './literals/list';
import listItemLiteral from './literals/listItem';
import titleLiteral from './literals/mainTitle';
import paragraphLiteral from './literals/paragraph';
import qLiteral from './literals/q';
import strongLiteral from './literals/strong';
import subtitleLiteral from './literals/subtitle';
import separatorLiteral from './literals/separator';
import blockquoteLiteral from './literals/blockquote';
import codeLiteral from './literals/code';
import delLiteral from './literals/del';

export {
  headingLiteral,
  imageLiteral,
  italicLiteral,
  linkLiteral,
  listLiteral,
  listItemLiteral,
  titleLiteral,
  paragraphLiteral,
  qLiteral,
  strongLiteral,
  subtitleLiteral,
  separatorLiteral,
//----------
  blockquoteLiteral,
  codeLiteral,
  delLiteral
};
