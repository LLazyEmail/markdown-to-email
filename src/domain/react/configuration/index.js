import {
    replaceMarkdown,
} from '../replace-markdown/replace-md';


// import Replacer from '../../parserMDReact/replace';

// function configureReplacer(state, isFull) {
//   // CAN WE MOVE OUT CALLBACKS INTO REPLACER?????
//   ReactReplacer.replaceMDBinded = replaceMarkdown.bind(state);

//   // I think we didnt use it quite often
//   // ReactReplacer.replaceMDBindedPreviewText = replaceMarkdownPreviewText.bind(state);

//   // NOTE DON'T CHANGE ORDER OF FUNCTION CALLS
//   if (isFull) {
//     ReactReplacer.previewText();
//   }

//   ReactReplacer.comments();
//   ReactReplacer.typography();

//   ReactReplacer.ulList();
//   // TODO UPDATE OLLIST CALLBACK, IT'S NOT WORKING
//   //   ReactReplacer.olList()

//   ReactReplacer.blockQuote();
//   ReactReplacer.hr();
//   ReactReplacer.paragraph();

//   // TODO CHECK CASES WHEN CONTENT CAN HAVE EMPTY LISTS OR BLOCKQUOTE
//   //   ReactReplacer.emptyUl();
//   //   ReactReplacer.emptyOl();
//   //   ReactReplacer.emptyBlockQuote();

//   ReactReplacer.br();
//   ReactReplacer.sponsorship();
//   ReactReplacer.mem();
//   ReactReplacer.separator();
// }

// export default configureReplacer;
