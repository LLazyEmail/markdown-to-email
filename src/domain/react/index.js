import mainObject from './pre-replace-object';
import configureReplacer from './configuration';
export { mainObject, configureReplacer };