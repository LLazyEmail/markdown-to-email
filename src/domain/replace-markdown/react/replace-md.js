// import debuggingReplacer from './debugging';
// import { replaceMarkdown } from '../../replace-class';

const prepOurCallback = (name, configurationMap) => {
  if (!name) {
    throw new Error(`name of ${name} is undefined or empty`);
  }

  const settings = configurationMap[name];

  if (!settings) {
    throw new Error(`object by name ${name} not found`);
  }

  return settings;
};

// if (debug) {
//   //--------------
//   if (!settings.literal) {
//     console.log(callbackName, settings.constant);
//     console.log('-----------------');
//   }
//   //---------------
// }
// TODO we finally can replaceMarkdown()*** for this one.

export function replaceMarkdownReact(callback_name) {
  // const debugging = new replaceMarkdownDebug(callback_name);

  const singleElement = prepOurCallback(callback_name, this.configurationMap);

  // --------- comment for debugging purposes
  // // --- You can comment this when you debugging our wrapper
  // const _replacer = singleElement.replacer.bind(this);
  // this.content = this.content.replace(singleElement.constant, _replacer);

  const _replacer = singleElement.replacer.bind(this);
  this.content = this.content.replace(singleElement.constant, _replacer);

  //   // -------------- Uncomment for debugging reasons
  // if (debuggingReplacer(callback_name)) {
  //   // console.log(callback_name);

  //   //     console.log(replacedString);

  //   //     // TODO another case why i dont like this solution
  //   //     // with string.replace - when you forget to return something it's hard to catch
  // }
  // return '';
  //-----------------------
}

// function replaceMarkdownDebug(callback_name) {
//   // -------------- Uncomment for debugging reasons
//   if (debuggingReplacer(callback_name)) {

//     console.log(callback_name);

//     const _replacer = singleElement.replacer.bind(this);

//     console.log(replacedString);

//     // TODO another case why i dont like this solution
//     // with string.replace - when you forget to return something it's hard to catch
//     this.content = this.content.replace(singleElement.constant, _replacer);
//   }
//   return '';
//   //--------------
// }

// TODO mayber replace it with class?
// class replaceMarkdownReact {
//   constructor(callback_name) {
//     const singleElement = prepOurCallback(callback_name, false);
//     // --------- comment for debugging purposes
//     // // --- You can comment this when you debugging our wrapper
//     // const _replacer = singleElement.replacer.bind(this);
//     // this.content = this.content.replace(singleElement.constant, _replacer);
//     //   // -------------- Uncomment for debugging reasons
//     if (debuggingReplacer(callback_name)) {

//       // console.log(callback_name);
//       const _replacer = singleElement.replacer.bind(this);

//       //     console.log(replacedString);
//       //     // TODO another case why i dont like this solution
//       //     // with string.replace - when you forget to return something it's hard to catch
//       this.content = this.content.replace(singleElement.constant, _replacer);

//     }
//     return '';
//     //   //-----------------------
//   }
// }
