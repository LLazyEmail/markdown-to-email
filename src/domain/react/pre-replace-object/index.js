import typography from 'atherdon-newsletter-react-layouts-typography';
import { Callbacks } from '../../../callbacksReact';

import {
  REGEXP_HEADER,
  REGEXP_IMAGE,
  REGEXP_LINK,
  REGEXP_STRONG,
  REGEXP_DEL,
  REGEXP_Q,
  REGEXP_CODE,
  // REGEXP_UL_LIST,
  // REGEXP_OL_LIST,
  REGEXP_BLOCKQUOTE,
  REGEXP_HR,
  REGEXP_BR,
  REGEXP_EM,
  // REGEXP_SPONSORSHIP,
  // REGEXP_MEM,
  // REGEXP_PREVIEW_TEXT,
  REGEXP_PARAGRAPH,
  // REGEXP_SEPARATOR,
} from '../../../constants/index';

import objectBuilder from '../../md/object-builder';

const {
  _strong,
  _link,
  _blockquote,
  //   _mem,
  _header,
  _italic,
  _del,
  _q,
  _code,
  _hr,
  //   _empty,
  //------------
  // _ulList,
  // _olList,
  //------------
  _image,
  _paragraph,
  // _sponsorship,
  _br,
  //------------
  // _separator,

  // _previewText,
} = Callbacks;

const {
  headingComponent,
  imageComponent,
  italicComponent,
  linkComponent,
  mainTitleComponent,
  paragraphComponent,
  strongComponent,
  subtitleComponent,
  qComponent,
  codeComponent,
} = typography;

const map = {
  strong: objectBuilder(REGEXP_STRONG, _strong, strongComponent),
  link: objectBuilder(REGEXP_LINK, _link, linkComponent),
  del: objectBuilder(REGEXP_DEL, _del, false),
  image: objectBuilder(REGEXP_IMAGE, _image, imageComponent),
  // TODO find out what module to use to import previewText
  // previewText: objectBuilder(
  //   REGEXP_PREVIEW_TEXT,
  //   _previewText,
  //   previewTextLiteral,
  // ),
  italic: objectBuilder(REGEXP_EM, _italic, italicComponent),
  header: objectBuilder(REGEXP_HEADER, _header, headingComponent),
  // TODO header arent working as suppose too
  subtitle: objectBuilder(REGEXP_HEADER, _header, subtitleComponent),
  title: objectBuilder(REGEXP_HEADER, _header, mainTitleComponent),
  q: objectBuilder(REGEXP_Q, _q, qComponent),
  code: objectBuilder(REGEXP_CODE, _code, codeComponent),

  // listItem:objectBuilder(
  //   REGEXP_SUB_LISTS,
  //   getParsedSubList,
  //   listItemLiteral
  //   ),

  // this object used only as a stupid way to add a parch for different cases of lists
  // first two params never used
  // list:objectBuilder(REGEXP_UL_LIST, getParsedSubList, listLiteral),
  // ulList:objectBuilder(REGEXP_UL_LIST, _ulList, listLiteral),
  // olList:objectBuilder(REGEXP_OL_LIST, _olList, false),
  blockquote: objectBuilder(REGEXP_BLOCKQUOTE, _blockquote, false),
  hr: objectBuilder(REGEXP_HR, _hr, false),
  paragraph: objectBuilder(REGEXP_PARAGRAPH, _paragraph, paragraphComponent),
  br: objectBuilder(REGEXP_BR, _br),
};
// sponsorship:objectBuilder(
//   REGEXP_SPONSORSHIP,
//   _sponsorship,
//   sponsorLiteral,
// );

// // //  memeLiteral,
// memes:objectBuilder(REGEXP_MEM, _meme, false);

// separator:objectBuilder(
//   REGEXP_SEPARATOR,
//   _separator,
//   separatorLiteral
//   );

// // 'REGEXP_EMPTY_UL': {
// //   constant: REGEXP_EMPTY_UL,
// //   replacer: emptyUl
// // },

// // 'REGEXP_EMPTY_OL': {
// //   constant: REGEXP_EMPTY_OL,
// //   replacer: emptyOl
// // },

// // 'REGEXP_EMPTY_BLOCKQUOTE': {
// //   constant: REGEXP_EMPTY_BLOCKQUOTE,
// //   replacer: emptyBlockQuote
// // },

export default map;
