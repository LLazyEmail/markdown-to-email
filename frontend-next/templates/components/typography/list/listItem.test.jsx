import React from 'react';
import { create } from 'react-test-renderer';
import ReactDOM from 'react-dom';
import { render } from '@testing-library/react';
import ListItem from './ListItem/Item';

describe('ListItem component', () => {
  test('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ListItem />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test('renders correctly', () => {
    const tree = create(<ListItem><h1>Some text</h1></ListItem>);
    expect(tree).toMatchSnapshot(); // .toJSON()
  });

  test('compoents props should contain children(h1) elements', () => {
    const component = create(<ListItem><h1>Some text</h1></ListItem>);
    const { root } = component;
    expect(root.props.children.type).toBe('h1');
  });

  test('<p> has correct children elements', () => {
    const { getByTestId } = render(<ListItem><h1>Hello</h1></ListItem>);
    const p = getByTestId('listItemTest');

    expect(p.children[0].textContent).toBe('Hello');
  });
});
