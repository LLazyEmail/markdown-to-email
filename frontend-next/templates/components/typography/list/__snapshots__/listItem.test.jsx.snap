// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem component renders correctly 1`] = `
<li
  style={
    Object {
      "WebkitTextSizeAdjust": "100%",
      "msTextSizeAdjust": "100%",
      "msoLineHeightRule": "exactly",
    }
  }
>
  <div
    data-testid="listItemTest"
    dir="ltr"
    role="presentation"
    style={
      Object {
        "WebkitTextSizeAdjust": "100%",
        "color": "#111111",
        "fontFamily": "\\"Source Sans Pro\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
        "fontSize": "18px",
        "lineHeight": "125%",
        "margin": "10px 0",
        "msTextSizeAdjust": "100%",
        "msoLineHeightRule": "exactly",
        "padding": 0,
        "textAlign": "left",
      }
    }
  >
    <h1>
      Some text
    </h1>
  </div>
</li>
`;
